packages/batch-submitter/src/batch-submitter/tx-batch-submitter.ts

```diff
@@ -802,9 +770,11 @@ export class TransactionBatchSubmitter extends BatchSubmitter {
     block.transactions[0].txType = txTypePlainText[block.transactions[0].txType]
     block.transactions[0].queueOrigin =
       queueOriginPlainText[block.transactions[0].queueOrigin]
-    // For now just set the l1BlockNumber based on the current l1 block number
     if (!block.transactions[0].l1BlockNumber) {
-      block.transactions[0].l1BlockNumber = this.lastL1BlockNumber
```

Comment:
:+1: Just to be clear, this was added because before, L2Geth would _not_ return inject the `l1BlockNumber` in the transaction's context?

Reply:
This is correct